ballerina/validator_visitor.bal

```diff
@@ -123,23 +123,30 @@ class ValidatorVisitor {
 
     isolated function visitInputObject(parser:ArgumentNode argumentNode, __InputValue schemaArg, string fieldName) {
         __Type argType = getOfType(schemaArg.'type);
-        __InputValue[] inputFields = <__InputValue[]>argType?.inputFields;
-        self.validateInputObjectFields(argumentNode, inputFields);
-        foreach __InputValue inputField in inputFields {
-            __Type subArgType = getOfType(inputField.'type);
-            __InputValue subInputValue = inputField;
-            if argumentNode.getValue().hasKey(inputField.name) {
-                parser:ArgumentValue|parser:ArgumentNode fieldValue = argumentNode.getValue().get(inputField.name);
-                if fieldValue is parser:ArgumentNode {
-                    self.visitArgument(fieldValue, {input:subInputValue, fieldName:fieldName});
-                }
-            } else {
-                if ((subInputValue.'type).kind == NON_NULL && schemaArg?.defaultValue is ()) {
-                    string message = string`Field "${fieldName}" argument "${subInputValue.name}" of type ` +
-                    string`"${getTypeNameFromType(subInputValue.'type)}" is required, but it was not provided.`;
-                    self.errors.push(getErrorDetailRecord(message, argumentNode.getLocation()));
+        __InputValue[]? inputFields = argType?.inputFields;
+        if inputFields is __InputValue[] {
+            self.validateInputObjectFields(argumentNode, inputFields);
+            foreach __InputValue inputField in inputFields {
+                __Type subArgType = getOfType(inputField.'type);
+                __InputValue subInputValue = inputField;
+                if argumentNode.getValue().hasKey(inputField.name) {
+                    parser:ArgumentValue|parser:ArgumentNode fieldValue = argumentNode.getValue().get(inputField.name);
+                    if fieldValue is parser:ArgumentNode {
+                        self.visitArgument(fieldValue, {input:subInputValue, fieldName:fieldName});
+                    }
+                } else {
+                    if ((subInputValue.'type).kind == NON_NULL && schemaArg?.defaultValue is ()) {
+                        string message = string`Field "${fieldName}" argument "${subInputValue.name}" of type ` +
+                        string`"${getTypeNameFromType(subInputValue.'type)}" is required, but it was not provided.`;
+                        self.errors.push(getErrorDetailRecord(message, argumentNode.getLocation()));
+                    }
                 }
             }
+        } else {
+            string expectedTypeName = getOfType(schemaArg.'type).name.toString();
+            string message = string`${expectedTypeName} cannot represent non ${expectedTypeName} value: {}`;
```

Comment:
Is this the correct way, or shall we use `null` here instead of `{}`?

Reply:
GraphQL playground is also used `{}` in error messages. shall we keep it as same? WDYT?